### ISSUE TITLE
associative array indexing in arithmetic contexts doesn't work

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#294]
Reported by @Crestwave in #290 .  I'm pretty sure fixing #287 will fix this.

```
$ declare -A arr
$ arr[0]=0
$ (( var = ${arr[0]} ))
$ (( var = arr[0] ))
*** Error has no source location info ***
fatal: Expected array in index expression, got {'0': '0'}
```

**Comment by andychu** [comment]
This seems to be fixed now.  But I might disallow associative arrays here because it doesn't work reliably in bash?

Or maybe just allow them with string indices?

**Comment by andychu** [comment]
Released with 0.7.pre1: https://www.oilshell.org/release/0.7.pre1/
